react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.rotateImages = this.rotateImages.bind(this);
+    this.changeImage = this.changeImage.bind(this);
+    this.shiftImage = this.shiftImage.bind(this);
+    this.state = {
+      currentImage: 0,
+      interval: setInterval(this.rotateImages, 3000)
```

Comment:
It's accepted "best practice" to do stuff like this in `componentDidMount`, in case there's an error during the first render. You don't want to leave an interval running if the component doesn't successfully get mounted.